department-of-veterans-affairs/caseflow
# Missing Ratings | Add current rating profile endpoint to Caseflow

Connects #14839 

### Description
This PR integrates the new current rating endpoint from ruby-bgs into Caseflow, and adds a new model for working with data returned by this endpoint.

### Acceptance Criteria
- [x] Add find_current_rating_profile_by_ptcpnt_id to BGSService
- [x] Add a CurrentRating model which can fetch the (single) current rating for a veteran
  - [x] The same "self" classes in the Rating model aren't needed, but the same instance methods are needed
- [x] Add to BGSService fakes

### Testing Plan
- New spec added: `current_rating_spec.rb`
- Testing locally
  - The fake rating store has a list of available participant IDs: `BGSService.rating_store.all_keys`
  - Use `CurrentRating.fetch_by_participant_id` to fetch an instance and play around with it
  - Check the behavior for a non-existent participant ID
- Testing in unmodified UAT (to confirm everything is safe) and prod (to see representative data)
  - Paste the contents of `current_rating.rb` into Rails console
  - Inside a `class ExternalApi::BGSService`/`end` block:
    - Paste the contents of the `def find_current_rating_profile_by_ptcpnt_id` definition
  - Use `CurrentRating.fetch_by_participant_id` with a bunch of `Veteran.participant_id`s and play around. The new test code can provide some inspiration.
  - Check the behavior for a non-existent participant ID

### Code Documentation Updates
